addons/hr_expense/models/ir_attachment.py

```diff
@@ -0,0 +1,29 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_attachments = super().create(vals_list)
+        for values in vals_list:
+            res_model = values.get('res_model', self.env.context.get('default_res_model'))
+            res_id = values.get('res_id', self.env.context.get('default_res_id'))
+            if res_model == 'hr.expense' and res_id:
+                for attachment in new_attachments:
```

Comment:
Instead, can't you just filter new_attachments on res_model = 'hr.expense'? :thinking: